auth/src/main/java/com/firebase/ui/auth/ui/phone/SpacedEditText.java

```diff
@@ -65,6 +65,11 @@ public void setText(CharSequence text, BufferType type) {
      */
     @Override
     public void setSelection(int index) {
+        // Prevent index from ever exceeding the bounds of the text (See #779)
+        if (index >= mOriginalText.length()) {
```

Comment:
I don't think that's going to work: the index was 11 (should be 10) in the crash, but the original text is only 5 chars long. TBH, I'm confused as to how any of this works... how is the span 21 characters long? As a very ugly solution, we could just use the catch block to call super with index 0. Or we could multiply `mOriginalText.length()` by 2.

Reply:
In the crash report we see `originalText=------` which is a 6 character long string of dashes.  We insert a space in between each pair (but not before first or after last) which gives an 11-character long string of `-_-_-_-_-_-`.

I have no idea how we ever got to 21 though... I think the most faithful thing to do is just select the very end if we get too large of an argument.